### PR TITLE
feat: add auth

### PR DESCRIPTION
**Pull request by fbwoolf**
This is working so it can be merged.

### CODE REVIEW

File: src/components/app.tsx [hunk 1]
```diff
@@ -1,21 +1,64 @@
...
 export const App: React.FC = memo(() => {
+  const [appPrivateKey, setAppPrivateKey] = useAtom(appPrivateKeyAtom);
+  const [authResponse, setAuthResponse] = useAtom(authResponseAtom);
+  const [user, setUser] = useAtom(userAtom);
+
+  useEffect(() => {
+    if (userSession.isUserSignedIn()) {
+      void setUser(user);
+    }
+  }, []);
```
Comment (aulneau) [inline src/components/app.tsx:22]: This seems unneeded. You're setting setUser with user.

you can also do this:
```ts
const isUserSignedInAtom = atom((get) => {
const userSession = get(userSessionAtom); // would need to make this
return userSession.isUserSignedIn();
});
```

but better yet, you'd just have something that is like this:

```ts
const userAtom = atomWithDefault((get) => { 
const userSession = get(userSessionAtom);
return userSession.loadUserData();
});
```

**Review by aulneau — APPROVED** [review]

**Comment by aulneau** [comment]
I'm going to merge this though so I can use some of this :) feel free to make changes in a follow up PR

File: src/components/app.tsx (hunk 1 above)
Comment (fbwoolf) [reply src/components/app.tsx:22]: Yeah, good point. I need to look it over a bit bc I was referencing the current test-app auth and there seems to be some code there that needs to be refactored too. I had stuff moved out in a `use-auth` hook but pulled it back in bc thought seemed unneeded. Let me refactor with your suggestion.

**Comment by fbwoolf** [comment]
> I'm going to merge this though so I can use some of this :) feel free to make changes in a follow up PR

Great, sounds good!

**Pull request merged**